### PR TITLE
fix: escape xml characters

### LINKED ISSUE #519: Escape invalid xml characters for all listing APIs
Currently Next Marker, Marker, Prefix and CommonPrefixes keys are not escaped in the xml. This results in invalid XML if object keys have invalid XML characters.

### PR DESCRIPTION
**Pull request by rahulreddy**
List parts does not need tests for prefix, marker, next marker as they are string
hashes and not object keys which may have invalid xml characters.

fixes #519

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 1]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with & in it\'s key',
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: not sure if you care, but the rest of your it statements use the backticks around the symbol and here they are missing

`it\'s` can just be `its`, the apostrophe is only used for contractions ('it is' --> "it's")

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 2]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with  in `\'` it\'s key',
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: think you swapped the order of the symbol and 'in' : should be `with \ in its key` (with backticks)

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 3]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with  in `<` it\'s key',
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: same for this one and below

**Comment by rahulreddy** [comment]
@ironman-machine try

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 4]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
         });
     });
 });
+
+/** Tests for special characters in XML **/
+
+/* eslint-disable no-param-reassign */
+function createParts(sigCfg, bucketUtil, s3, key) {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: createPart

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 5]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with in `<` its key',
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: minor:  with ~in~ \`<\` in

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 6]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with in `\'` its key',
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: minor:  with ~in~ \`\\` in

File: tests/functional/aws-node-sdk/test/object/listParts.js [hunk 7]
```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
...
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with in `>` its key',
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/object/listParts.js]: minor:  with ~in~ \`>\` in

**Pull request merged**